mule-extensions-api-persistence/src/main/java/org/mule/runtime/extension/internal/persistence/ModelPropertyMapTypeAdapter.java

```diff
@@ -73,8 +72,8 @@ public void write(JsonWriter out, HierarchyClassMap<ModelProperty> modelProperty
   }
 
   @Override
-  public HierarchyClassMap<ModelProperty> read(JsonReader in) throws IOException {
-    final HierarchyClassMap<ModelProperty> modelPropertyHashMap = new HierarchyClassMap<>();
+  public Map<Class<? extends ModelProperty>, ModelProperty> read(JsonReader in) throws IOException {
+    final Map<Class<? extends ModelProperty>, ModelProperty> modelPropertyHashMap = new HashMap<>();
```

Comment:
this needs to be a LinkedHashMap. Otherwise, all the model tests will become flaky.